pkg/apiserver/middlewares/v1/api_key.go

```diff
@@ -139,8 +139,13 @@ func (a *APIKey) MiddlewareFunc() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		var bouncer *ent.Bouncer
 
+		clientIP := c.ClientIP()
+		if clientIP == "" && isUnixSocket(c) {
```

Comment:
But what if the request is proxied to unix socket from a web server? we need to set the remote addr on the request before c.ClientIP() is called.

https://github.com/gotify/server/blob/46281d6a51dc0ca6a2ac13f81db89224b990d876/router/router.go#L34-L39